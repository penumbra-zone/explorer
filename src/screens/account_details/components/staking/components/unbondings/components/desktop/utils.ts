export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 25,
  },
  {
    key: 'commission',
    align: 'right',
    width: 25,
  },
  {
    key: 'amount',
    width: 25,
    align: 'right',
  },
  {
    key: 'linkedUntil',
    width: 25,
    align: 'right',
  },
];
