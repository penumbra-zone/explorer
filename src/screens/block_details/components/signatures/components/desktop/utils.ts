export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 34,
  },
  {
    key: 'votingPower',
    align: 'right',
    width: 33,
  },
  {
    key: 'signed',
    width: 33,
  },
];
