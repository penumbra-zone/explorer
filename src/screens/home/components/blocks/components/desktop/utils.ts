export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
}[] = [
  {
    key: 'height',
  },
  {
    key: 'validator',
  },
  {
    key: 'hash',
  },
  {
    key: 'txs',
  },
  {
    key: 'time',
    align: 'right',
  },
];
