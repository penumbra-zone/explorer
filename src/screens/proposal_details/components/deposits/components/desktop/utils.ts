export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'depositor',
    width: 34,
  },
  {
    key: 'amount',
    width: 33,
    align: 'right',
  },
  {
    key: 'time',
    width: 33,
  },
];
