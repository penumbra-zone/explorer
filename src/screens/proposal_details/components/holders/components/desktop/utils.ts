export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 10,
  },
  {
    key: 'address',
    width: 30,
  },
  {
    key: 'quantity',
    width: 20,
    align: 'right',
  },
  {
    key: 'percentage',
    width: 20,
  },
  {
    key: 'value',
    align: 'right',
    width: 20,
  },
];
