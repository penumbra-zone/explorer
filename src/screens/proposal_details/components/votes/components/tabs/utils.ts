export const tabLabels = [
  {
    key: 'all',
    num: 30,
  },
  {
    key: 'yes',
    num: 3,
  },
  {
    key: 'no',
    num: 3,
  },
  {
    key: 'noWithVeto',
    num: 3,
  },
];
