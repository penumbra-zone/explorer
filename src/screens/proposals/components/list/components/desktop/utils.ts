export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'idx',
    width: 15,
  },
  {
    key: 'token',
    width: 10,
  },
  {
    key: 'price',
    align: 'right',
    width: 15,
  },
  {
    key: 'change',
    align: 'right',
    width: 15,
  },
  {
    key: 'volume',
    align: 'right',
    width: 15,
  },
  {
    key: 'marketCap',
    align: 'right',
    width: 15,
  },
  {
    key: 'holders',
    align: 'right',
    width: 15,
  },
];
