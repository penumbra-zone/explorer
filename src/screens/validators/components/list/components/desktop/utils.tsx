import React from 'react';
import {
  Typography,
} from '@material-ui/core';
import { InfoPopover } from '@components';

export const fetchColumns = (t): {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: React.ReactNode;
  sort?: boolean;
}[] => {
  return ([
    {
      key: 'idx',
      width: 10,
    },
    {
      key: 'validator',
      width: 20,
      sort: true,
    },
    {
      key: 'stake',
      align: 'right',
      width: 15,
      sort: true,
    },
    {
      key: 'fee',
      align: 'right',
      width: 10,
      sort: true,
    },
    {
      key: 'lastVote',
      align: 'right',
      width: 15,
    },
    {
      key: 'skipRate',
      width: 15,
      sort: true,
    },
    {
      key: 'condition',
      align: 'center',
      width: 15,
      component: (
        <Typography variant="h4" className="label popover">
          {t('condition')}
          <InfoPopover
            content={t('conditionExplanation')}
          />
        </Typography>
      ),
    },
  ]);
};
