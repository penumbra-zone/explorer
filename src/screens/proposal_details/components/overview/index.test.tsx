import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Overview from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  BoxDetails: (props) => <div id="BoxDetails" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: BlockDetails/Overview', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Overview />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
