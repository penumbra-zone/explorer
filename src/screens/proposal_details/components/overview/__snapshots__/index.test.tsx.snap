// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: BlockDetails/Overview matches snapshot 1`] = `
<div
  details={
    Array [
      Object {
        "className": "makeStyles-copyText",
        "detail": <React.Fragment>
          <SvgIconCopy
            onClick={[Function]}
          />
          <WithStyles(ForwardRef(Typography))
            className="value"
            variant="body1"
          >
            HjTgC7ukwMA1gsE...E6B6X
          </WithStyles(ForwardRef(Typography))>
        </React.Fragment>,
        "label": "tokens:address",
      },
      Object {
        "detail": "7",
        "label": "tokens:decimals",
      },
      Object {
        "className": "makeStyles-copyText",
        "detail": <React.Fragment>
          <SvgIconCopy
            onClick={[Function]}
          />
          <Link
            href="/accounts/123"
            passHref={true}
          >
            <WithStyles(ForwardRef(Typography))
              className="value"
              component="a"
              variant="body1"
            >
              HjTgC7ukwMA1gsE...E6B6X
            </WithStyles(ForwardRef(Typography))>
          </Link>
        </React.Fragment>,
        "label": "tokens:mintAuthority",
      },
      Object {
        "className": "makeStyles-copyText",
        "detail": <React.Fragment>
          <SvgIconCopy
            onClick={[Function]}
          />
          <Link
            href="/accounts/123"
            passHref={true}
          >
            <WithStyles(ForwardRef(Typography))
              className="value"
              component="a"
              variant="body1"
            >
              HjTgC7ukwMA1gsE...E6B6X
            </WithStyles(ForwardRef(Typography))>
          </Link>
        </React.Fragment>,
        "label": "tokens:freezeAuthority",
      },
      Object {
        "detail": <WithStyles(ForwardRef(Typography))
          className="value"
          component="a"
          href="tokenname.com"
          rel="noopener"
          target="_blank"
          variant="body1"
        >
          tokenname.com
        </WithStyles(ForwardRef(Typography))>,
        "label": "tokens:officialSite",
      },
    ]
  }
  id="BoxDetails"
  title="tokens:overview"
/>
`;
